### PR TITLE
Handle exception - first implementation

### PR DESCRIPTION
**Pull request by dennisseah**
test implementation on infra scaffold command. please provide feedback.

### CODE REVIEW

File: src/lib/errorStatusCode.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// please do not change the status code numbers
+// you can add new ones but not changing the existing ones
```
Comment (andrebriggs) [inline src/lib/errorStatusCode.ts:2]: @dennisseah I assume the guidance is to add increasing error status codes with spaces in between to give ourselves room to add related error statuses?

**Review by andrebriggs — APPROVED** [review]

**Comment by andrebriggs** [comment]
@dennisseah it might help to expand on [this](https://github.com/CatalystCode/spk/blob/master/guides/command-implementation.md#error-messages) section with the concrete implementation we have here. This way we can codify this pattern

File: src/lib/errorStatusCode.ts (hunk 1 above)
Comment (dennisseah) [reply src/lib/errorStatusCode.ts:2]: precisely

**Comment by dennisseah** [comment]
> @dennisseah it might help to expand on [this](https://github.com/CatalystCode/spk/blob/master/guides/command-implementation.md#error-messages) section with the concrete implementation we have here. This way we can codify this pattern

I would have to update my design doc once I have collected all feedbacks on this PR and then I will update that section.  hope that this is ok with you.

File: src/lib/errorStatusCode.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export const errorStatusCode = {
+  1000: "command fails to execute",
```
Comment (edaena) [inline src/lib/errorStatusCode.ts]: nit: the numbers can also be mapped to an enum or a constat, VALIDATION_ERROR, EXECUTION_ERROR, ENVIRONMENT_ERROR, GIT_ERROR, COMMAND_FAIL_ERROR, or something similar

Comment (dennisseah) [reply src/lib/errorStatusCode.ts]: accept. will make the change. thanks

**Review by NathanielRose — APPROVED** [review]
Unit and Int tests work. LGTM

**Review by yradsmikham — APPROVED** [review]

**Review by edaena — APPROVED** [review]

**Pull request merged**